src/mlpack/methods/dbscan/dbscan_impl.hpp

```diff
@@ -68,19 +68,31 @@ size_t DBSCAN<RangeSearchType, PointSelectionPolicy>::Cluster(
   centroids.zeros(data.n_rows, numClusters);
 
   // Calculate number of points in each cluster.
-  arma::Row<size_t> counts;
-  counts.zeros(numClusters);
-  for (size_t i = 0; i < data.n_cols; ++i)
+  arma::Row<size_t> counts(numClusters);
+
+  #pragma omp parallel
   {
-    if (assignments[i] != SIZE_MAX)
+    MatType localCentroids(data.n_rows, numClusters);
+    arma::Row<size_t> localCounts(numClusters);
+
+    #pragma omp for nowait
+    for (size_t i = 0; i < data.n_cols; ++i)
     {
-      centroids.col(assignments[i]) += data.col(i);
-      ++counts[assignments[i]];
+      if (assignments[i] != SIZE_MAX)
+      {
+        localCentroids.col(assignments[i]) += data.col(i);
+        ++localCounts[assignments[i]];
+      }
+    }
+
+    #pragma omp critical
+    {
+      centroids += localCentroids;
+      counts += localCounts;
     }
   }
 
-  // We should be guaranteed that the number of clusters is always greater than
-  // zero.
+  // Normalize centroids
```

Comment:
Can you revert to or include the intent of the original comment? Someone wrote that there for a reason.